PLOS/wombat
# AMBR-905 Fix jackson dependencies for new spring version

*JIRA issue:* https://jira.plos.org/jira/browse/AMBR-905

## What this PR does:

Upgrades jackson to a version compatible with our spring version.

The underlying stack trace here was being hidden by an error in our error handler. I was able to get to the original stack trace by commenting out the exception handler.

The underlying error was:
```
org.springframework.web.util.NestedServletException: Handler dispatch failed; nested exception is java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.ObjectWriter.forType(Lcom/fasterxml/jackson/databind/JavaType;)Lcom/fasterxml/jackson/databind/ObjectWriter;
```

## Developer Notes

The core of this fix is in 308eeeeb8.

The rest is test cleanup, and adding a regression test.

## To test:

Checkout a969d5f0e and run `mvn clean test`. You should get a test failure - this is expected, because the test here is to ensure that we do not regress this problem with the taxonomy browser.

Now checkout 308eeeeb8, and run `mvn clean test`. The tests should pass.

## Further test notes

I believe the file https://github.com/PLOS/wombat/blob/308eeeeb85c763765bee15b72f22aeef49bab46e/src/test/java/org/ambraproject/wombat/controller/TaxonomyControllerTest.java can provide an exemplar for tests going forward.

Note the configuration inner class: https://github.com/PLOS/wombat/blob/308eeeeb85c763765bee15b72f22aeef49bab46e/src/test/java/org/ambraproject/wombat/controller/TaxonomyControllerTest.java#L52 which sets up a single, non-mocked controller. The rest of the mocked beans are provided by the `ControllerTest` superclass. (In the future we could abstract out more of this into a more generic class.) The required instances are set up by Spring with `@Autowired`. I hope it makes sense. I reworked most of the other controller tests as well using this pattern.

# Code Author Tasks:

> This list should be finished before code review:
- [x] Ticket AC is updated with any decisions made in comments or side conversations.
- [x] Testing or PO Accept notes that will assist the tester, Product Owner or reviewer are set in the ticket.
- [x] All PRs are created and linked for work I have done against other projects as part of this ticket.
- [x] Intergration Tests have been addressed.  QA has been notified of the feature and a ticket has been created for that work, or integration tests have been updated. 
- [x] There are unit tests sufficient for both positive and negative test cases, test coverage has not decreased as part of this work. 
- [x] Any necessary documentation in confluence or READMEs is updated.
__
# Code Reviewer Tasks
- [x] I ran the code (in the review environment or locally). I agree the running code fulfills the Acceptance Criteria as stated on the JIRA ticket. If not practical I have seen some evidence that the code does what it says (tests have passed).

## Project or Language Specific Tasks ###
